### PR TITLE
EVG-3535 small trigger refactor

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: model/trigger/task.go [hunk 1]
```diff
@@ -440,15 +438,11 @@ func (t *taskTriggers) taskRuntimeChange(sub *event.Subscription) (*notification
...
-		return nil, nil
-	}
-	percentChange := math.Abs(100*ratio - 100)
-	if percentChange < percent {
+	shouldNotify, percentChange := runtimeExceedsThreshold(percent, prevTaskDuration, thisTaskDuration)
+	if !shouldNotify {
 		return nil, nil
 	}
-	return t.generate(sub, fmt.Sprintf("changed in runtime by %f%% (over threshold of %f%%)", percentChange, percent))
+	return t.generate(sub, fmt.Sprintf("changed in runtime by %.1f%% (over threshold of %.1f%%)", percentChange, percent))
```
Comment (richardsamuels) [inline model/trigger/task.go]: Nit, but I'd advocate for using the original string representation of the float (percentString) for the second template parameter.

File: model/trigger/task.go [hunk 2]
```diff
@@ -440,15 +438,11 @@ func (t *taskTriggers) taskRuntimeChange(sub *event.Subscription) (*notification
 	}
```
Comment (richardsamuels) [inline model/trigger/task.go]: Should we also extract the logic in taskExceedsDuration and similar triggers as well?

Comment (john-m-liu) [reply model/trigger/task.go]: I originally planned to do that as well, but there are enough small differences that extracting this doesn't make it any more succinct

File: model/trigger/task.go (hunk 1 above)
Comment (john-m-liu) [reply model/trigger/task.go]: done

**Pull request merged**